DataDog/test-infra-definitions
# [windows] allow auth keys reset on windows instances

What does this PR do?
---------------------

* refactor ssh setup script to allow authorized keys reset on Windows instances
* move script to components to re-use it across providers
* pass arguments as params on Azure
* replace arguments in script on AWS

Which scenarios this will impact?
-------------------

Windows on Azure and AWS

Motivation
----------

This is blocking the Windows Dev Env, as it uses a custom AMI that has ssh configured.

Additional Notes
----------------

This PR is best reviewed by commit.

To test it, use 
```bash
inv aws.create-vm --ami-id=ami-0007ea7a7f8fa0afc --os-family=windows --architecture=x86_64 --no-install-agent
```

And ensure you can connect through ssh. The ami is on `agent-sandbox` account.

More details about aws userdata: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html
